server/routes/v1/deployment.ts

```diff
@@ -161,3 +166,67 @@ export const resetDeploymentApprovals = [
     return res.json(deployment)
   },
 ]
+
+export const fetchRawModelFiles = [
+  ensureUserRole('user'),
+  bodyParser.json(),
+  async (req: Request, res: Response) => {
+    const { uuid, version, fileType } = req.params
+    const deployment = await findDeploymentByUuid(req.user!, uuid)
+
+    if (deployment === null) {
+      throw NotFound({ deploymentUuid: uuid }, `Unable to find deployment for uuid ${uuid}`)
+    }
+
+    if (!req.user._id.equals(deployment.owner)) {
+      throw Unauthorised(
+        { deploymentOwner: deployment.owner },
+        `User is not authorised to download this file. Requester: ${req.user._id}, owner: ${deployment.owner}`
+      )
+    }
+
+    if (deployment.managerApproved !== 'Accepted') {
+      throw Unauthorised(
+        { approvalStatus: deployment.managerApproved },
+        'User is not authorised to download this file as it has not been approved.'
+      )
+    }
+
+    if (fileType !== 'code' && fileType !== 'binary') {
+      throw NotFound({ fileType }, 'Unknown file type specificed')
+    }
+
+    const versionDocument: VersionDoc | null = await findVersionByName(req.user!, deployment.model, version)
+    const bucketName: string = config.get('minio.uploadBucket')
+    const client = new Minio.Client(config.get('minio'))
+
+    if (versionDocument === null) {
+      throw NotFound({ versionId: version }, `Unable to find version for id ${version}`)
+    }
+
+    let filePath
+
+    if (fileType === 'code') {
+      filePath = versionDocument.files.rawCodePath
+    }
+    if (fileType === 'binary') {
+      filePath = versionDocument.files.rawBinaryPath
+    }
+
+    // Stat object to get size so browser can determine progress
+    const { size } = await client.statObject(bucketName, filePath)
+
+    //res.set('Content-Disposition', contentDisposition(fileType, { type: 'inline' }))
+    res.set('Content-disposition', `attachment; filename=${fileType}.zip`)
+    res.set('Content-Type', 'application/zip')
+    res.set('Cache-Control', 'private, max-age=604800, immutable')
+    res.set('Content-Length', size.toString())
+    res.writeHead(200)
+
+    const stream: Readable = await client.getObject(bucketName, filePath)
```

Comment:
This is an invalid type, it should be `Readable | null`, but again you should be able to omit it entirely:

```javascript
const stream = await client.getObject(...)
```